vm/systemSmartContracts/validator.go

```diff
@@ -649,6 +649,10 @@ func (v *validatorSC) registerBLSKeys(
 	}
 
 	for _, blsKey := range newKeys {
+		if v.isNumberOfNodesTooHigh(registrationData) {
```

Comment:
why do we have to put the check in the for loop? Can it be taken outside?

Reply:
registrationData.BLSKeys is changed inside the for on line 675

Reply:
:man_facepalming: